go/arrow/array/string.go

```diff
@@ -246,7 +249,7 @@ func (a *LargeString) setData(data *Data) {
 
 	if vdata := data.buffers[2]; vdata != nil {
 		b := vdata.Bytes()
-		a.values = *(*string)(unsafe.Pointer(&b))
+		a.values = string(b[:])
```

Comment:
we shouldn't copy the entire buffer here. Keep the `unsafe.Pointer` here so we only perform copies in the `Value` method and only on the values that are requested.